source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Encode.netcoreapp.cs

```diff
@@ -170,18 +180,69 @@ private void Avx2Encode<T>(ref byte src, ref T dest, int sourceLength, int destL
 
                 // Load at c-4, as required by enc_reshuffle
                 src.AssertRead<Vector256<sbyte>, byte>(ref srcStart, sourceLength);
-                str = src.ReadVector256();
+                avxStr = src.ReadVector256();
             }
+#if DEBUG
+            Avx2Encoded?.Invoke();
+#endif
+            src = ref Unsafe.Add(ref src, 4);
+
+        Sse:
+            simdSrcEnd = ref Unsafe.Add(ref srcStart, (IntPtr)((uint)sourceLength - 16 + 1));   //  +1 for <
+
+            if (!Unsafe.IsAddressLessThan(ref src, ref simdSrcEnd))
+                goto Exit;
+
+            // Reuse AVX vectors
+            Vector128<sbyte>  shuffleVec          = avxShuffleVec.GetUpper();
```

Comment:
This creates a
```asm
vextracti128 ymm0, ymm0, 1
```
maybe it's better to just load the vector from ROS as the latency is quite high.

Reply:
Gutfeeling: leave it as is.